src/resolver.cpp

```diff
@@ -168,54 +211,69 @@ UsdOpenAssetIOResolver::~UsdOpenAssetIOResolver() {
 
 std::string UsdOpenAssetIOResolver::_CreateIdentifier(
     const std::string &assetPath, const ArResolvedPath &anchorAssetPath) const {
-  std::string identifier;
-
-  if (manager_->isEntityReferenceString(assetPath)) {
-    // If assetPath is an entity reference we must preserve it
-    // unmodified as the "identifier", since it'll be passed to
-    // subsequent member functions.  We assume it will (eventually)
-    // resolve to an absolute path, making the anchorAssetPath redundant
-    // (for now).
-    identifier = assetPath;
-  } else {
-    identifier = ArDefaultResolver::_CreateIdentifier(
-        assetPath, ArResolvedPath{locationInManagerContextForEntity(manager_, readContext_,
-                                                                    anchorAssetPath)});
-  }
-
-  logger_->debug("OPENASSETIO_RESOLVER: " + TF_FUNC_NAME());
-  logger_->debug("  assetPath: " + assetPath);
-  logger_->debug("  anchorAssetPath: " + anchorAssetPath.GetPathString());
-  logger_->debug("  result: " + identifier);
-
-  return identifier;
+  const std::string fnName = TF_FUNC_NAME();
```

Comment:
[Consider] Rather than capturing a name variable, we could just log the function name on entry and avoid the extra state. If we use the `catchAndLogExceptions` return value, I think we can avoid mixing the logic and the logging altogether.

A good formulation might be something like

```cpp
logger_->debug("OPENASSETIO_RESOLVER: " + TF_FUNC_NAME())
logger_->debug("  assetPath: " + assetPath);
logger_->debug("  anchorAssetPath: " + anchorAssetPath.GetPathString());

auto result = catchAndLogExceptions(...){...}

logger_->debug("  result: " + identifier);
```

That might give you scope to do some early returns in some if/else blocks in the implementation. It's quite nice actually, creates a seperation between system-logic and behaviour loggin.

Reply:
I like it. My only concern is multithreading, where log lines can be interleaved, so you don't know for certain which method caused the exception.

Reply:
[fix] Btw, David - I think you highlighted that Log message should be multi-line anyway, not split into separate calls? Now we're using `logger` not `TF` how about just dumping it into one entry?

Reply:
How does 81adf29 look?

Reply:
mar vel ouse

Reply:
Looks great.